urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/core/launcher/LauncherApplet.java

```diff
@@ -0,0 +1,156 @@
+import java.awt.HeadlessException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import com.kokakiwi.mclauncher.api.LauncherAPI;
+
+public class LauncherApplet extends Applet implements AppletStub
+{
+    private static final long         serialVersionUID = -2866153864231108251L;
+    
+    private final LauncherAPI         api;
+    
+    private final Map<String, String> params           = new LinkedHashMap<String, String>();
+    
+    private Applet                    applet;
+    
+    private boolean                   active           = false;
+    private int                       context          = 0;
+    
+    public LauncherApplet(LauncherAPI api) throws HeadlessException
+    {
+        this.api = api;
+        
+        params.put("stand-alone", "true");
+        params.put("username", api.getLoginer().getLastLogin().getUsername());
+        params.put("sessionid",
+                String.valueOf(api.getLoginer().getLastLogin().getSessionId()));
+        
+        setBackground(Color.black);
+    }
+    
+    public void appletResize(int width, int height)
+    {
+        
+    }
+    
+    public void replace(Applet applet)
+    {
+        this.applet = applet;
+        applet.setStub(this);
+        applet.setSize(getWidth(), getHeight());
+        
+        setLayout(new BorderLayout());
+        add(applet, "Center");
+        
+        applet.init();
+        active = true;
+        applet.start();
+        validate();
+    }
+    
+    public boolean isActive()
+    {
+        if (context == 0)
+        {
+            context = -1;
+            try
+            {
+                if (getAppletContext() != null)
+                {
+                    context = 1;
+                }
+            }
+            catch (final Exception localException)
+            {
+            }
+        }
+        if (context == -1)
+        {
+            return active;
+        }
+        return super.isActive();
+    }
+    
+    public String getParameter(String name)
+    {
+        String custom = api.getConfig().getString(name);
+        if (custom == null)
+        {
+            custom = params.get(name);
+            if (custom == null)
+            {
+                try
+                {
+                    custom = super.getParameter(name);
+                }
+                catch (final Exception e)
+                {
+                    params.put(name, null);
+                }
+            }
+        }
+        
+        System.out.println("Minecraft want param '" + name + "' = '" + custom
+                + "'");
+        
+        return custom;
+    }
+    
+    public URL getDocumentBase()
+    {
+        try
+        {
+            return new URL(api.getConfig()
+                    .getString("launcher.documentBaseURL"));
+        }
+        catch (final MalformedURLException e)
+        {
+            e.printStackTrace();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Stack trace information reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).